devito/propagator.py

```diff
@@ -200,7 +201,7 @@ def sympy_to_cgen(self, stencils):
                 # TODO: add support for double precision
                 self.add_local_var(name, np.float32)
                 # TODO: undo precision enforcing
-                factors.append(cgen.Assign(name, str(ccode(expr.xreplace(self._var_map))).replace("pow", "powf").replace("fabs", "fabsf")))
+                factors.append(cgen.Assign(name, str(ccode(self.time_substitutions(expr).xreplace(self._var_map))).replace("pow", "powf").replace("fabs", "fabsf")))
```

Comment:
This change seems unusually controversial for @navjotk and @mloubout. @navjotk is stating that it shouldn't work and it's causing him problems (details?) while @mloubout states this works (indeed tea is is green) and passes the adjoint test - our strongest test. 

@navjotk - I'm afraid that it seems like it's up to you to figure out why this is giving all the appearances of working while you are expecting it to break; and why is it that when it is coded in the way you expect it doesn't give the expected result.